here-naksha-lib-view/src/test/java/com/here/naksha/lib/view/PsqlViewTests.java

```diff
@@ -0,0 +1,337 @@
+    assertNotNull(schema);
+    return schema;
+  }
+
+  private String schema;
+
+  @Test
+  @Order(10)
+  @EnabledIf("runTest")
+  void createStorage() {
+    storage = new PsqlStorage(config);
+    schema = storage.getSchema();
+    if (!schema.equals(schema())) {
+      storage.setSchema(schema());
+    }
+    // Enable this code line to get debug output from the database!
+    // storage.setLogLevel(EPsqlLogLevel.DEBUG);
+  }
+
+  @Test
+  @Order(11)
+  @EnabledIf("dropInitially")
+  void dropSchemaIfExists() {
+    assertNotNull(storage);
+    storage.dropSchema();
+  }
+
+  @Test
+  @Order(13)
+  @EnabledIf("runTest")
+  void initStorage() {
+    assertNotNull(storage);
+    storage.initStorage(new Params().pg_hint_plan(false).pg_stat_statements(false));
+  }
+
+  @Test
+  @Order(20)
+  @EnabledIf("runTest")
+  void startWriteSession() {
+    assertNotNull(storage);
+    session = storage.newWriteSession(nakshaContext, true);
+    assertNotNull(session);
+  }
+
+  @Test
+  @Order(30)
+  @EnabledIf("runTest")
+  void createCollection() throws NoCursor {
+    assertNotNull(storage);
+    assertNotNull(session);
+    final WriteXyzCollections request = new WriteXyzCollections();
+    request.add(EWriteOp.CREATE, new XyzCollection(COLLECTION_0, false, false, true));
+    request.add(EWriteOp.CREATE, new XyzCollection(COLLECTION_1, false, false, true));
+    request.add(EWriteOp.CREATE, new XyzCollection(COLLECTION_2, false, false, true));
+    try (final ForwardCursor<XyzCollection, XyzCollectionCodec> cursor =
+             session.execute(request).getXyzCollectionCursor()) {
+      assertNotNull(cursor);
+      assertTrue(cursor.hasNext());
+    } finally {
+      session.commit(true);
+    }
+  }
+
+  @Test
+  @Order(31)
+  @EnabledIf("runTest")
+  void addFeatures() {
+    assertNotNull(storage);
+    assertNotNull(session);
+    PsqlFeatureGenerator fg = new PsqlFeatureGenerator();
+    final WriteXyzFeatures requestTest0 = new WriteXyzFeatures(COLLECTION_0);
+    final WriteXyzFeatures requestTest1 = new WriteXyzFeatures(COLLECTION_1);
+    final WriteXyzFeatures requestTest2 = new WriteXyzFeatures(COLLECTION_2);
+    for (int i = 0; i < 10; i++) {
+      final XyzFeature feature = fg.newRandomFeature();
+      feature.setGeometry(new XyzPoint(0d, 0d));
+      requestTest0.add(EWriteOp.PUT, feature);
+
+      XyzFeature featureEdited1 = feature.deepClone();
+      featureEdited1.setGeometry(new XyzPoint(1d, 1d));
+      requestTest1.add(EWriteOp.PUT, featureEdited1);
+
+      XyzFeature featureEdited2 = feature.deepClone();
+      featureEdited2.setGeometry(new XyzPoint(2d, 2d));
+      requestTest2.add(EWriteOp.PUT, featureEdited2);
+    }
+
+    try {
+      session.execute(requestTest0);
+      session.execute(requestTest1);
+      session.execute(requestTest2);
+    } finally {
+      session.commit(true);
+    }
+  }
+
+
+  @Test
+  @Order(41)
+  @EnabledIf("runTest")
+  void viewQueryTest_pickTopLayerResult() throws NoCursor {
```

Comment:
I think (and suggest), it will be valuable adding/enhancing tests considering the main moderation use-case:
1. Use of IgnoreMissingResolver when main request is getById(s).
2. Use of 3 layers, where the features are available in intermediate and lowest layer (but not in top layer). Especially for Spatial requests (BBox / Tile).

Reply:
Ad 1. This is something very useful, so I changed the implementation, now view-lib verifies if the request is by IDs only - if it is, then it omits second query. Now, the clients don't have to remember to use IgnoreMissingResolver when they use IDs only